### PR TITLE
[FEATURE] GCS `DataConnector` schema validation

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- https://superconductive.atlassian.net/secure/RapidBoard.jspa?rapidView=24&projectKey=GREAT&modal=detail&selectedIssue=GREAT-101
- Update YAML schema validation and `Asset` attrs to allow for GCS-specific terminology. This will later be refactored in GREAT-89 but is necessary in order to get the initial implementations of the GCS `DataConnectors` up and running.

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/data_context/types/base.py [hunk 1]
```diff
@@ -505,6 +543,21 @@ def validate_schema(self, data, **kwargs):
...
+            "gcs_options" in data or "bucket_or_name" in data or "max_results" in data
+        ) and not (
+            data["class_name"]
+            in [
+                "InferredAssetGCSDataConnector",
+                "ConfiguredAssetGCSDataConnector",
+            ]
+        ):
+            raise ge_exceptions.InvalidConfigError(
+                f"""Your current configuration uses one or more keys in a data connector, that are required only by an
```
Comment (alexsherstinsky) [inline great_expectations/data_context/types/base.py]: @cdkini Minor: "an" -> "a".  Thanks!

Comment (alexsherstinsky) [inline great_expectations/data_context/types/base.py]: @cdkini either delete the comma or use "which" in place of "that".  Thanks!

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
Strong work -- please correct the grammar/etc. and good to go!

File: great_expectations/data_context/types/base.py (hunk 1 above)
Comment (cdkini) [reply great_expectations/data_context/types/base.py]: Fixed! Good eye 👀

Comment (cdkini) [reply great_expectations/data_context/types/base.py]: Changed all instances of this in the file!

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Review by Shinnnyshinshin — APPROVED** [review]
Looks wonderful :) thank you

**Review by fjork3 — APPROVED** [review]
woo GCS!

**Pull request merged**